virttest/qemu_vm.py

```diff
@@ -896,25 +896,35 @@ def add_memorys(devices, params):
 
             :param devices: VM devices container
             """
-            options = []
+            options, devs = [], []
+            params.setdefault("mem", "512")
             mem_params = params.object_params("mem")
-            if mem_params.get("maxmem"):
+            mem_params.setdefault("automem", "yes")
+            automem = mem_params["automem"] == "yes"
+            if automem:
```

Comment:
Requesting guest memory size bigger than the value of `maxmem` in the qemu `-m` option is a negative test case, and in this situation, a VM create error is what we expected.

We can let user decide to shrink the memory size to be close to the free value, but if we got a very small value (such as 256 MB), the guest os is more likely to refuse to boot up. I don't know if setting this as default is a good choice.

Moreover, if the guest is using a memory file backend (hugetlbfs), I'm not sure if these code can calculate the value correctly.